### ISSUE TITLE
Dart code only runs once

### ISSUE THREAD

**Issue by exo-pla-net** [dart-lang/webdev#647]
Using the "web-simple" scaffold from stagehand, and running webdev serve, the browser will display:

> "Your Dart app is running."

as expected.

However, upon hitting refresh on the browser, nothing will display. No console errors, nothing. The html itself is still being served, but no dart code appears to be executing.

I'm running Dart VM version: 2.5.0-dev.2.1 (Thu Aug 15 16:05:39 2019 +0200) on "macos_x64", and webdev 2.5.0.

**Comment by exo-pla-net** [comment]
This issue appears to be particular to Chrome (no extensions). Refreshing works in Safari.

Chrome: Version 76.0.3809.132 (Official Build) (64-bit)

**Comment by hamidrezabstn** [comment]
same issue here with firefox and chrome both

**Comment by Hexer10** [comment]
I guess it is a duplicate of https://github.com/dart-lang/webdev/issues/610 , try to enable persistent logging in the chrome devtools console, and check if it returns an error like this:

```
Unhandled error detected in the injected client.js script.

You can disable this script in webdev by passing --no-injected-client if it
is preventing your app from loading, but note that this will also prevent
all debugging and hot reload/restart functionality from working.

The original error is below, please file an issue at
https://github.com/dart-lang/webdev/issues/new and attach this output:

Instance of 'Event'
null

Navigated to http://<host>:8080/
Hello world!
```

I had fixed this by downgrading to dart 2.2

**Comment by hamidrezabstn** [comment]
@exo-pla-net @Hexer10 
Downgrading to webdev 2.3.0 would work as well

**Comment by exo-pla-net** [comment]
Indeed, this is a duplicate: I get the same error using persistent logging upon refresh.

Downgrading webdev to 2.3.0 resolves the problem. To save someone a search, install it as so:
`pub global activate webdev 2.3.0`

**Comment by jakemac53** [comment]
We have tracked this issue down and should have it fixed soon. You can also work around the issue by enabling debugging (with `--debug`).

**Comment by johnpryan** [comment]
This might be unrelated, but using the `--debug` flag and refreshing quickly can cause a crash:

```
Unhandled exception:
Bad state: Could not connect to application with appInstanceId: 011de960-d351-11e9-fa56-07fc5041e708
#0      DevHandler.startDebugService (package:dwds/src/handlers/dev_handler.dart:108:7)
<asynchronous suspension>
#1      DevHandler.loadAppServices.<anonymous closure> (package:dwds/src/handlers/dev_handler.dart:137:19)
<asynchronous suspension>
#2      __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:281:23)
#3      DevHandler.loadAppServices (package:dwds/src/handlers/dev_handler.dart:135:24)
#4      DevHandler._handleConnection.<anonymous closure> (package:dwds/src/handlers/dev_handler.dart:265:22)
<asynchronous suspension>
#5      _rootRunUnary (dart:async/zone.dart:1136:13)
#6      _RootZone.runUnaryGuarded (dart:async/zone.dart:1317:7)
#7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#8      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#9      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#10     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#11     _rootRun (dart:async/zone.dart:1120:38)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#15     _rootRun (dart:async/zone.dart:1124:13)
#16     _CustomZone.run (dart:async/zone.dart:1021:19)
#17     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#18     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#19     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#20     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#21     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#22     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

Another temporary workaround is to use `pub run build_runner serve` instead.

**Comment by grouma** [comment]
Thanks for filing these issues! 

I have published a new version of webdev `2.5.1` which should resolve these issues. If you experience further problems please continue to open issues.
